utils/global-state-update-gen/src/auction_utils.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use casper_engine_test_support::internal::LmdbWasmTestBuilder;
+use casper_execution_engine::shared::stored_value::StoredValue;
+use casper_types::{
+    system::auction::{Bid, SeigniorageRecipient, SeigniorageRecipientsSnapshot},
+    AsymmetricType, EraId, Key, PublicKey, U512,
+};
+
+use crate::utils::ValidatorsDiff;
+
+/// Generates a new `SeigniorageRecipientsSnapshot` based on:
+/// - The list of validators, in format (validator_public_key,stake), both expressed as strings.
+/// - The starting era ID (the era ID at which the snapshot should start).
+/// - Count - the number of eras to be included in the snapshot.
+pub fn gen_snapshot(
+    validators: Vec<(String, String)>,
+    starting_era_id: EraId,
+    count: u64,
+) -> SeigniorageRecipientsSnapshot {
+    let mut new_snapshot = BTreeMap::new();
+    for era_id in starting_era_id.iter(count) {
+        let mut era_validators = BTreeMap::new();
+        for (key_str, bonded_str) in &validators {
+            let key = PublicKey::from_hex(key_str.as_bytes()).unwrap();
+            let bonded_amount = U512::from_dec_str(bonded_str).unwrap();
+            let seigniorage_recipient =
+                SeigniorageRecipient::new(bonded_amount, Default::default(), Default::default());
+            let _ = era_validators.insert(key, seigniorage_recipient);
+        }
+        let _ = new_snapshot.insert(era_id, era_validators);
+    }
+
+    new_snapshot
+}
+
+/// Returns the set of public keys that have bids larger than the smallest bid among the new
+/// validators.
+pub fn find_large_bids(
+    builder: &mut LmdbWasmTestBuilder,
+    new_snapshot: &SeigniorageRecipientsSnapshot,
+) -> BTreeSet<PublicKey> {
+    let max_bid = new_snapshot
```

Comment:
The comment of the method says it's going to be the smallest bid but this is called `max_bid`. You're also taking the `min()` value later on.